streams.yml

```diff
@@ -69,6 +69,26 @@ rhel-8-golang:
   upstream_image_mirror:
   - registry.ci.openshift.org/ocp-private/builder-priv:rhel-8-golang-1.22-openshift-{MAJOR}.{MINOR}
 
+partner-rhel-8-golang-1.21:
+  image: openshift/golang-builder:v1.21.11-202407021847.g1ac3e39.el8
+  mirror: true
+  upstream_image: quay.io/openshift-release-dev/golang-builder--partner-share:rhel-8-golang-1.21-openshift-{MAJOR}.{MINOR}
```

Comment:
Could we add these destinations to the existing IBM stream by adding `upstream_image_mirror`` instead of making completely new streams to independently maintain? Until we need to differentiate what we push, `upstream_image_mirror` should work.